api/inc/vmpu_exports.h

```diff
@@ -189,6 +219,11 @@ typedef struct {
      * This is set to `NULL` by uVisor, signalling to the user lib that the
      * box heap needs to be initialized before use! */
     void * active_heap;
+
+    /* Box IDs */
+    int box_id_self;
+    int box_id_caller;
```

Comment:
This isn't thread-safe!

If I have two threads doing `rpc_fncall_waitfor` and they get scheduled during an RPC call, but before `uvisor_box_id_caller()` has been called, the `box_id_caller` may be overwritten by the RPC call serviced by the other thread. This variable would have to be moved into thread local storage.